### ISSUE TITLE
Bug in Total price calculation when using discount code to upgrade an already active account

### ISSUE THREAD

**Issue by mojtaba** [restrictcontentpro/restrict-content#42]
When the user uses the discount code to upgrade the account, the payment amount doesn't calculated correctly. The customer can get extra discount by upgrading the membership level by level, instead of paying for higher membership in the first time.

Suppose that we have a monthly level for 7$, a 3-month level for 14$ and a yearly level for 28$. We have also a 30 percent discount code.

A user buys a monthly account using the discount code, and pays 4.9$ instead of 7$. No problem so far. Then at the same day, she uses the same discount code, to upgrade the membership to 3-month level. Her total payment is calculated as follows:
(14 * 0.7) - 7 = 2.8$
As you see, she paid 7.7$ in total. If she buy the same membership in first payment without any upgrade, she had to pay 9.8$ in total. by far, she took 15% more discount with a simple trick and can get even more, by upgrading to yearly membership.

This is because of the wrong calculation formula:
`Total Today = (price * (1 - discount)) - proration`

The correct formula should be:
`Total Today = (price - proration) * (1 - discount)`

In other words, the discount should be applied after proration, not before.

Note: I know that I can enable the 'One Time' checkbox for the discount code. But that's not the solution, because I want my old users with expired memberships to be able to use this discount code.
